src/sensor/ping.h

```diff
@@ -85,6 +85,9 @@ class Ping : public Sensor
 
     Q_INVOKABLE void firmwareUpdate(QString fileUrl);
 
+    Q_INVOKABLE void stopDetector() override;
```

Comment:
Do these need to be QINVOKABLE?
Should they be QINVOKABLE?

Reply:
Yes, they will be used into the qml here https://github.com/bluerobotics/ping-components/pull/81.

The idea is to allow the user to enable auto detect or connect using manual mode.

Reply:
It feels like the Ping object only should be calling these methods, and we could have a QProperty for enabling autodetect

Reply:
Yes, but to do such thing, `connectLink` need to be defined here in Ping to call stopDetect. 
or do you want to create something like:
```cpp
Q_PROPERTY(bool autodetect READ _autodetect WRITE setAutoDetect NOTIFY autoDetectUpdate)
```
And inside setAutoDetect do the stop and startDetector ?

Reply:
Yes, that's the idea. 

I was thinking it can go here https://github.com/bluerobotics/ping-protocol-cpp/blob/dev/src/sensor/sensor.cpp#L19

or like
```
Ping::connectLink() {
  detector.stop();
  Sensor::connectLink();
}
```